server/src/features/FoldingRangeRequestHandler.ts

```diff
@@ -0,0 +1,29 @@
+import { FoldingRange, FoldingRangeParams } from 'vscode-languageserver';
+
+import { YmlDefinitionProvider } from '../definitions';
+import { AbstractYmlObject } from '../yml-objects';
+
+export function foldingRangeRequestHandler(definitionsProvider: YmlDefinitionProvider) {
```

Comment:
Kinda funny that we say we’re about to handle a folding request or whatever, and only get weird YML definitions as an argument.
Actually, it seems that this function registers the ranges of all the things that could be folded, instead of trying to fold a particular item. Not what I expected from its name xd
Also curious (and related) that you went through the trouble of adding in-code comments but didn’t add a dedicated documentation for the function itself.